common/src/express/router.ts

```diff
@@ -0,0 +1,118 @@
+import { Schema, Get } from '../api-types'
+import { PathParams } from 'express-serve-static-core'
+import express from 'express'
+import { TypedHandler, TypedRouter } from './types'
+
+class TypedAPIError extends Error {}
+class FrozenRouterError extends TypedAPIError {
+  constructor() {
+    super('Cannot modify a frozen router.')
+  }
+}
+// @ts-ignore unused
+class UnhandledPathError extends TypedAPIError {
+  // TODO: add a check to the router to raise this error when appropriate
+  constructor(unhandledPaths: Set<Schema.Path>) {
+    const pathList = Array.from(unhandledPaths).join(', ')
+    super(`Router cannot be frozen without handling the following paths: ${pathList}`)
+  }
+}
+
+export class Router<Endpoint extends Schema.Endpoint> implements TypedRouter<Endpoint> {
+  protected router: express.Router
+  private frozen: boolean = false
+
+  constructor(protected endpoint: Endpoint, options?: express.RouterOptions) {
+    this.router = express.Router(options)
+  }
+
+  public freeze(): this {
+    if (this.frozen) throw new FrozenRouterError()
+    this.frozen = true
+    return this
+  }
```

Comment:
I'm trying to model (in runtime) what we might eventually do in compile time. After you freeze a router, you shouldn't be able to modify it. A file might define a `clubRouter` and export `clubRouter.freeze()`, so that other files that import it can't modify it.